release/lightning_tests/workloads/test_trainer.py

```diff
@@ -0,0 +1,138 @@
+        avg_acc = torch.stack([x["val_accuracy"] for x in outputs]).mean()
+        self.log("ptl/val_loss", avg_loss, sync_dist=True)
+        self.log("ptl/val_accuracy", avg_acc, sync_dist=True)
+
+    def configure_optimizers(self):
+        optimizer = torch.optim.Adam(self.parameters(), lr=self.lr)
+        return optimizer
+
+
+class MNISTDataModule(pl.LightningDataModule):
+    def __init__(self, batch_size=100):
+        super().__init__()
+        self.data_dir = os.getcwd()
+        self.batch_size = batch_size
+        self.transform = transforms.Compose(
+            [transforms.ToTensor(), transforms.Normalize((0.1307,), (0.3081,))]
+        )
+
+    def setup(self, stage=None):
+        # split data into train and val sets
+        if stage == "fit" or stage is None:
+            mnist = MNIST(
+                self.data_dir, train=True, download=True, transform=self.transform
+            )
+            self.mnist_train, self.mnist_val = random_split(mnist, [55000, 5000])
+
+        # assign test set for use in dataloader(s)
+        if stage == "test" or stage is None:
+            self.mnist_test = MNIST(
+                self.data_dir, train=False, download=True, transform=self.transform
+            )
+
+    def train_dataloader(self):
+        return DataLoader(self.mnist_train, batch_size=self.batch_size, num_workers=4)
+
+    def val_dataloader(self):
+        return DataLoader(self.mnist_val, batch_size=self.batch_size, num_workers=4)
+
+    def test_dataloader(self):
+        return DataLoader(self.mnist_test, batch_size=self.batch_size, num_workers=4)
+
+
+def get_mnist_dataloaders():
+    transform = transforms.Compose(
+        [transforms.ToTensor(), transforms.Normalize((0.1307,), (0.3081,))]
+    )
+    mnist_train = MNIST("/tmp/data", train=True, download=True, transform=transform)
+    mnist_val = MNIST("/tmp/data", train=False, download=True, transform=transform)
+    train_loader = DataLoader(mnist_train, batch_size=32, shuffle=True)
+    val_loader = DataLoader(mnist_val, batch_size=32, shuffle=False)
+    return train_loader, val_loader
+
+
+if __name__ == "__main__":
+    start = time.time()
+
+    lightning_config = (
+        LightningConfigBuilder()
+        .module(MNISTClassifier, feature_dim=128, lr=0.001)
+        .trainer(
+            max_epochs=3,
+            accelerator="gpu",
+            logger=CSVLogger("logs", name="my_exp_name"),
+        )
+        .fit_params(datamodule=MNISTDataModule(batch_size=128))
+        .checkpointing(monitor="ptl/val_accuracy", mode="max", save_last=True)
+        .build()
+    )
+
+    scaling_config = ScalingConfig(
+        num_workers=3, use_gpu=True, resources_per_worker={"CPU": 1, "GPU": 1}
+    )
+
+    trainer = LightningTrainer(
+        lightning_config=lightning_config,
+        scaling_config=scaling_config,
+    )
+
+    trainer.fit()
+
+    taken = time.time() - start
+    result = {
+        "time_taken": taken,
```

Comment:
Can we report some metrics about the training as part of job output? Like accuracy.